scripts/upload_gatk_docs/upload_gatk_tools_docs.py

```diff
@@ -0,0 +1,238 @@
+    """Get the url to post an article to the zendesk folder for this gatk release."""
+    url = 'https://gatk.zendesk.com/api/v2/help_center/en-us/sections.json'
+    response = requests.get(url, auth=(username + '/token', token))
+    response = response.json()
+
+    section_exists = False
+    for section in response['sections']:  # look for an existing folder with this gatk version name
+        if section['name'] == gatk_version:
+            url = section['url']
+            section_exists = True
+    if not section_exists:  # if it doesn't already exist, create a folder for this gatk version in Tool Index
+        url_Tool_Index = 'https://gatk.zendesk.com/api/v2/help_center/en-us/categories/360002369672/sections.json'
+
+        json_file = {"section": {"position": 2,
+                                 "sorting": "title",  # will sort contents alphabetically by article title
+                                 "locale": "en-us",
+                                 "name": gatk_version,
+                                 "description": "Tool documentation for GATK release " + gatk_version
+                                 }}
+
+        data = json.dumps(json_file)
+
+        try:
+            # create the folder
+            response = requests.post(url_Tool_Index, headers=headers, data=data, auth=(username + '/token', token))
+        except: 
+            exit("ERROR: Can't make section")
+
+        url = response.json()['section']['url']  # folder url
+
+    # edit url to be the API call to post an article inside
+    post_section_url = url.replace('.json', '/articles.json')
+
+    return post_section_url
+
+
+def post_articles(gatk_version, article_dict, username, token, headers, gatk_team_id, permission_group_id, user_id):
+    """Post articles to Zendesk."""
+    # get the post url (API call) for the GATK version section folder (e.g. '4.1.4.0') (create the folder if it doesn't exist yet)
+    post_section_url = post_section(gatk_version, username, token, headers)
+
+    # first loop thorugh and post all articles, collecting their urls
+    for article in article_dict.values():  # each `article` is a GatkDocs class
+        # post articles in that section folder
+        article.post(post_section_url, user_id, gatk_team_id, permission_group_id, headers, username, token)  # this sets article.url as well
+
+
+def update_articles(article_dict, gatkdoc_version, headers, username, token, updatefolder=None):
+    """Update articles with correct article links."""
+    # Creating the folder for the updated files
+    if updatefolder is None:
+        updatefolder = 'gatkdoc_update_' + gatkdoc_version
+    if not os.path.exists(updatefolder):
+        os.makedirs(updatefolder)
+
+    # Get each article
+    for article in article_dict.values():
+        print('in original article ' + article.file_name)
+        with open(article.local_path, 'r') as clean_file:
+            clean_file_data_soup = bs(clean_file, 'html.parser')
```

Comment:
could you add a comment here explaining what this line is doing? additionally perhaps leave `BeautifulSoup` named that way when you import it, or name it something else more descriptive?